### PR TITLE
feat: add GitHub Org Repository sync to insight page creation

### PR DESCRIPTION
**Pull request by brandonroberts**
## Description

- Adds option to sync a GitHub Organization Repos into an insight page during creation

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Closes https://github.com/open-sauced/engineering/issues/187

## Mobile & Desktop Screenshots/Recordings

insight-repo-sync

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Review by bdougie — APPROVED** [review]
I really like this quality of life improvement. We need more of this in the app. While exploring this through QA, I noticed a familiar face in the contribution list for twitter/the-algorithm.

On the profile it is hard to know which contribution is for the algorithm without clicking each link. It would be nice to filter per repo or highlight on hover or similar. 

Screen Shot 2023-10-31 at 11 21 52 AM

File: components/organisms/InsightPage/InsightPage.tsx [hunk 1]
```diff
@@ -38,6 +38,12 @@ enum RepoLookupError {
   Error = 4,
 }
 
+enum OrgLookupError {
+  Initial = 0,
+  Invalid = 1,
+  Error = 2,
+}
```
Comment (nickytonline) [inline components/organisms/InsightPage/InsightPage.tsx:45]: nit: Use a [const enum](https://www.typescriptlang.org/docs/handbook/enums.html#const-enums).

File: components/organisms/InsightPage/InsightPage.tsx [hunk 2]
```diff
@@ -383,6 +395,18 @@ const InsightPage = ({ edit, insight, pageRepos }: InsightPageProps) => {
...
+  const getOrganizationLookupError = (code: OrgLookupError) => {
+    if (code === OrgLookupError.Error) {
+      return <Error errorMessage="There was error retrieving this organization's public repositories." />;
+    }
+
+    if (code === OrgLookupError.Invalid) {
+      return <Error errorMessage="This organization entered is invalid." />;
+    }
+
+    return <></>;
```
Comment (nickytonline) [inline components/organisms/InsightPage/InsightPage.tsx:407]: refactor (blocking): return null

```suggestion
    return null;
```

**Review by nickytonline — APPROVED** [review]
Thanks for making the changes @brandonroberts! :shipit:

**Review by OgDev-01 — APPROVED** [review]
🍕

**Pull request merged**